modules/apps/change-tracking/change-tracking-rest-impl/src/main/java/com/liferay/change/tracking/rest/internal/resource/v1_0/EntryResourceImpl.java

```diff
@@ -14,17 +14,154 @@
+import com.liferay.portal.kernel.dao.orm.QueryDefinition;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+import com.liferay.portal.vulcan.util.SearchUtil;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.component.annotations.ServiceScope;
 
 /**
  * @author Máté Thurzó
+ * @author Zoltan Csaszi
  */
 @Component(
 	properties = "OSGI-INF/liferay/rest/v1_0/entry.properties",
 	scope = ServiceScope.PROTOTYPE, service = EntryResource.class
 )
 public class EntryResourceImpl extends BaseEntryResourceImpl {
+
+	@Override
+	public Page<Entry> getCollectionEntriesPage(
+			Long collectionId, String changeTypesFilter,
+			String classNameIdsFilter, Boolean collision, String groupIdsFilter,
+			Integer status, String userIdsFilter, Pagination pagination,
+			Sort[] sorts)
+		throws Exception {
+
+		CTCollection ctCollection = _ctCollectionLocalService.fetchCTCollection(
+			GetterUtil.getLong(collectionId));
+
+		if (ctCollection == null) {
+			throw new IllegalArgumentException(
+				"Unable to get change tacking collection " + collectionId);
+		}
+
+		collision = GetterUtil.getBoolean(collision);
+
+		QueryDefinition<CTEntry> queryDefinition =
+			SearchUtil.getQueryDefinition(CTEntry.class, pagination, sorts);
+
+		if (status != null) {
+			queryDefinition.setStatus(status);
+		}
+		else {
+			queryDefinition.setStatus(WorkflowConstants.STATUS_DRAFT);
+		}
+
+		List<CTEntry> ctEntries = _ctEngineManager.getCTEntries(
+			ctCollection,
+			GetterUtil.getLongValues(_getFiltersArray(groupIdsFilter)),
+			GetterUtil.getLongValues(_getFiltersArray(userIdsFilter)),
+			GetterUtil.getLongValues(_getFiltersArray(classNameIdsFilter)),
+			GetterUtil.getIntegerValues(_getFiltersArray(changeTypesFilter)),
+			collision, queryDefinition);
+
+		int totalCount = _ctEngineManager.getCTEntriesCount(
+			ctCollection,
+			GetterUtil.getLongValues(_getFiltersArray(groupIdsFilter)),
+			GetterUtil.getLongValues(_getFiltersArray(userIdsFilter)),
+			GetterUtil.getLongValues(_getFiltersArray(classNameIdsFilter)),
+			GetterUtil.getIntegerValues(_getFiltersArray(changeTypesFilter)),
+			collision, queryDefinition);
+
+		List<Entry> entries = _getEntries(ctEntries);
+
+		return Page.of(entries, pagination, totalCount);
+	}
+
+	private List<Entry> _getEntries(List<CTEntry> ctEntries) {
+		Stream<CTEntry> stream = ctEntries.stream();
+
+		return stream.map(
+			this::_toEntry
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private String[] _getFiltersArray(String filterString) {
```

Comment:
mmm can you use arrays instead? they are defined as several parameters &classNameId=20123&classNameId=23434 but they are more standard and converted automatically by JAX-RS implementations